framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -221,6 +221,11 @@ protected boolean isPrimarySubtype(
             AnnotatedTypeMirror subtype, AnnotatedTypeMirror supertype, boolean annosCanBeEmtpy) {
         final AnnotationMirror subtypeAnno = subtype.getAnnotationInHierarchy(currentTop);
         final AnnotationMirror supertypeAnno = supertype.getAnnotationInHierarchy(currentTop);
+        if (checker.getTypeFactory().hasQualifierParameterInHierarchy(supertype, currentTop)
+                && checker.getTypeFactory().hasQualifierParameterInHierarchy(subtype, currentTop)) {
+            return isAnnoSubtype(subtypeAnno, supertypeAnno, annosCanBeEmtpy)
```

Comment:
Isn't this equivalent to `areSame`? Can you add a comment what is different, maybe whether one can be null?